src/Factory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Dhii\Services;
+
+use Dhii\Services\Factories\Constructor;
+use Psr\Container\ContainerInterface;
+
+/**
+ * A simple implementation for a factory service.
+ *
+ * This implementation will automatically resolve any specified dependencies and pass them as arguments to the
+ * definition function. The container will NOT be included in the arguments.
+ *
+ * Example usage:
+ * ```
+ * new Factory(['foo', 'bar'], function($foo, $bar) {
+ *      return new SomeClass($foo, $bar);
+ * });
+ * ```
+ *
+ * @since [*next-version*]
+ * @see   Constructor For a similar implementation that automatically injects dependencies into constructors.
+ * @see   Extension For a similar implementation that can be used with extension services.
+ */
+class Factory extends Service
+{
+    /**
+     * @since [*next-version*]
+     *
+     * @var callable
+     */
+    protected $definition;
+
+    /**
+     * @inheritDoc
+     *
+     * @since [*next-version*]
+     *
+     * @param callable $definition The factory definition.
+     */
+    public function __construct(array $dependencies, callable $definition)
+    {
+        parent::__construct($dependencies);
+
+        $this->definition = $definition;
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke(ContainerInterface $c)
+    {
+        $deps = Service::resolveKeys($c, $this->dependencies);
+
+        return ($this->definition)(...$deps);
```

Comment:
This appears to automatically support PHP 8's upcoming [named arguments][] feature! ❤️ 

[named arguments]: https://stitcher.io/blog/php-8-named-arguments

Reply:
Not necessarily. The contents of `$deps` are keyed according to the dependency array that is passed to `Factory`. So this would only be true if the dep keys are the same as the definition parameter names.

Works:
```
new Function(['foo', 'bar'], function($foo, $bar) {});
```

Doesn't:
```
new Function(['some/foo', 'other/bar'], function($foo, $bar) {});
```

I wonder if this will break in PHP 8. If it does, the keys will need to be stripped from `$deps` to ensure that the dependencies are passed to the definition ordinally rather than canonically.

Reply:
I would contend that it is a good idea to simply resolve the dependencies according to the key order, populate `$deps` in the right order together with the keys, and let it work when it works.